python/cudf/cudf/core/indexing.py

```diff
@@ -266,11 +266,41 @@ def _getitem_tuple_arg(self, arg):
             else:
                 return columns_df.index._get_row_major(columns_df, arg[0])
         else:
-            df = DataFrame()
-            for col in columns_df.columns:
-                # need Series() in case a scalar is returned
-                df[col] = Series(columns_df[col].loc[arg[0]])
-            df.columns = columns_df.columns
+            if isinstance(arg[0], slice):
+                start_index, stop_index = columns_df.index.find_label_range(
+                    arg[0].start, arg[0].stop
+                )
+
+                pos_slice = slice(start_index, stop_index, arg[0].step)
+                df = columns_df[pos_slice]
+            else:
+                tmp_arg = arg
+                if is_scalar(arg[0]):
+                    # If a scalar, there is possibility of having duplicates.
+                    # Join would get all the duplicates. So, coverting it to
+                    # a array kind.
+                    tmp_arg = ([tmp_arg[0]], tmp_arg[1])
+                if len(tmp_arg[0]) == 0:
+                    return columns_df[0:0]
+                tmp_arg = (column.as_column(tmp_arg[0]), tmp_arg[1])
+
+                if pd.api.types.is_bool_dtype(tmp_arg[0]):
+                    df = columns_df._apply_boolean_mask(tmp_arg[0])
+                else:
+                    tmp_col_name = str(uuid4())
```

Comment:
Generating a random name here feels extremely gross. I would say we could just take advantage of suffixes but then we go through an expensive renaming process instead, so lets leave this for now.